apps/management/lib/management/view_state.ex

```diff
@@ -0,0 +1,62 @@
+defmodule Management.ViewState do
+  @moduledoc """
+  View state management behaviour and default implementation on a per-collection
+  basis. `ViewState` collections should track only one type of object.
+
+  ## Configuration
+
+  Configure `ViewState` with `use/2` macro.
+
+  * `instance` - Brook instance name. See `Brook.instance()` for more info.
+  * `collection` - Name for object grouping. See `Brook.view_collection()` for more info.
+  """
+
+  @callback collection() :: Brook.view_collection()
+  @callback persist(Brook.view_key(), map) :: :ok
+  @callback get(Brook.view_key()) :: {:ok, map} | {:error, term}
+  @callback get_all() :: {:ok, [map]} | {:error, term}
+  @callback delete(Brook.view_key()) :: :ok
+
+  defmacro __using__(opts) do
+    instance = Keyword.fetch!(opts, :instance)
+    collection = Keyword.fetch!(opts, :collection)
+
+    quote location: :keep do
+      @behaviour Management.ViewState
+
+      @impl true
+      def collection do
+        unquote(collection)
+      end
+
+      @impl true
+      def persist(key, object) do
+        unquote(collection)
+        |> Brook.ViewState.merge(key, object)
+      end
+
+      @impl true
+      def get(key) do
+        unquote(instance)
+        |> Brook.get!(unquote(collection), key)
```

Comment:
why are you calling `Brook.get!` here and piping to `Ok.ok` but below you're calling `Brook.get_all` without the `!`?

Reply:
So `Brook.get` and `Brook.get!` do the same thing. They either return `{:ok, _}` or raise an exception. Seems like a miss on Brook's part.

I use `get!` here to denote that an exception will be raised. `get` would hide that.

Reply:
`Brook.get!` _shouldn't_ be returning an :ok tuple. Under the hood `Brook.get!` is just calling `Brook.get` and unwrapping the tuple, but if you're using `Brook.get!` and getting back an {:ok, _} then there's a nested :ok tuple being stored.

Reply:
Yeah, sorry, it's the error path that's the same across `get` and `get!`. My thought process still applies though. I wanted to be explicit about that error behavior so I'm using `get!` for now.